### ISSUE TITLE
Docker Rootless Ubuntu 22.04 - Unable to apply cgroup configuration (`...: Interactive authentication required.: unknown.`)

### ISSUE THREAD

**Issue by yghazi** [moby/moby#45014]
### Description

I'm using Ubuntu 22.04 and docker version 23.0.1 to setup `docker-rootless`. The original docker setup works out. But the trouble is with the rootless version. I can do `docker ps` after and it works as expected. But when I try to run `docker run hello-world` to see if the installation actually works, it says:

```
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to
 start container process: unable to apply cgroup configuration: unable to start unit "docker-
17cf4f4cf9e70ec40048541524c9ce751f3e1a1ec87074b2a746ebdaad9fa2cd.scope" (properties [{Name:Description 
Value:"libcontainer container 17cf4f4cf9e70ec40048541524c9ce751f3e1a1ec87074b2a746ebdaad9fa2cd"} {Name:Slice 
Value:"user.slice"} {Name:Delegate Value:true} {Name:PIDs Value:@au [33419]} {Name:MemoryAccounting Value:true} 
{Name:CPUAccounting Value:true} {Name:IOAccounting Value:true} {Name:TasksAccounting Value:true} 
{Name:DefaultDependencies Value:false}]): Interactive authentication required.: unknown.
ERRO[0010] error waiting for container: 
```

### Reproduce

I've used [the official documentation](https://docs.docker.com/engine/security/rootless/) for setup. When I run `docker run hello-world`, the abovementioned error occurs. And I can't find a way to troubleshoot in the documentation.

### Expected behavior

docker run hello-world should run the hello-world container and exit with:

```
Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (amd64)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://hub.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/get-started/
```

### docker version

```bash
Client:
 Version:           23.0.0
 API version:       1.42
 Go version:        go1.19.5
 Git commit:        e92dd87
 Built:             Wed Feb  1 17:43:29 2023
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          23.0.1
  API version:      1.42 (minimum version 1.12)
  Go version:       go1.19.5
  Git commit:       bc3805a
  Built:            Thu Feb  9 19:47:01 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.6.16
  GitCommit:        31aa4358a36870b21a992d3ad2bef29e1d693bec
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
 rootlesskit:
  Version:          1.1.0
  ApiVersion:       1.1.1
  NetworkDriver:    slirp4netns
  PortDriver:       builtin
  StateDir:         /tmp/rootlesskit2752381057
 slirp4netns:
  Version:          1.0.1
  GitCommit:        6a7b16babc95b6a3056b33fb45b74a6f62262dd4
```

### docker info

```bash
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.16.0
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.23.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 3
  Running: 0
  Paused: 0
  Stopped: 3
 Images: 2
 Server Version: 23.0.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: false
  userxattr: true
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 31aa4358a36870b21a992d3ad2bef29e1d693bec
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  rootless
  cgroupns
 Kernel Version: 5.15.0-25-generic
 Operating System: Ubuntu 22.04.1 LTS (error determining if containerized)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.765GiB
 Name: vmi598327.contaboserver.net
 ID: bc893c37-facb-4cf5-b2d7-e4585c1cf741
 Docker Root Dir: /home/admin/.local/share/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No cpu cfs quota support
WARNING: No cpu cfs period support
WARNING: No cpu shares support
WARNING: No cpuset support
WARNING: No io.weight support
WARNING: No io.weight (per device) support
WARNING: No io.max (rbps) support
WARNING: No io.max (wbps) support
WARNING: No io.max (riops) support
WARNING: No io.max (wiops) support
```

### Additional Info

I've tried the same thing on 20.04 with the same results.

**Comment by AkihiroSuda** [comment]
Please try `sudo apt-get install -y dbus-user-session` and reboot

**Comment by yghazi** [comment]
Already did that to no avail.

**Comment by AkihiroSuda** [comment]
Is this new in v23? Was v20.10 working on your environment?

**Comment by AkihiroSuda** [comment]
Can't repro the issue locally on a clean installation of Ubuntu 22.04

**Comment by yghazi** [comment]
I haven't run v20.10 but I can try and share the results here. By the way similar situation with debian: https://serverfault.com/questions/1122063/rootless-docker-fails-with-systemd-error-interactive-authentication-required?noredirect=1#comment1466424_1122063

**Comment by AkihiroSuda** [comment]
Can you try this?


```bash
sudo mkdir -p /etc/systemd/system/user@.service.d

cat <<EOF | sudo tee /etc/systemd/system/user@.service.d/delegate.conf
[Service]
Delegate=cpu cpuset io memory pids
EOF

sudo systemctl daemon-reload

sudo reboot
```

**Comment by yghazi** [comment]
I've done that already. Also, on a clean 22.04 installation, 20.10 works:

```
Client: Docker Engine - Community
 Version:           20.10.23
 API version:       1.41
 Go version:        go1.18.10
 Git commit:        7155243
 Built:             Thu Jan 19 17:45:08 2023
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.23
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.10
  Git commit:       6051f14
  Built:            Thu Jan 19 17:42:57 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.18
  GitCommit:        2456e983eb9e37e47538f59ea18f2043c9a73640
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by yghazi** [comment]
UPDATE: I tried with an entirely clean 22.04 installation and the latest (23.0.1) also worked. I think the difference is in some automated hardening scripts that were changing the sysctl configs. Not sure which one is causing docker rootless not to work, though.

**Comment by yghazi** [comment]
Further update on this: it works fine if we update grub with `GRUB_CMDLINE_LINUX="systemd.unified_cgroup_hierarchy=0"` but then it also gives the warning: 
```
WARNING: Running in rootless-mode without cgroups. To enable cgroups in rootless-mode, you need to boot the
system in cgroup v2 mode.
```
If we switch the value of `systemd.unified_cgroup_hierarchy=0` to `systemd.unified_cgroup_hierarchy=1` the warning goes away since now we're using cgroup2, but it gives the same Interactive authentication required error.

**Comment by cristianrgreco** [comment]
Also running into this, rootless Docker in Github actions, worked on Ubuntu 20.04, no longer working on Ubuntu 22.04. Docker version: 23.0.0, OS: Ubuntu 22.04.2 LTS, Arch: x86_64.

Getting the following errors:

> Error: (HTTP code 500) server error - failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: unable to apply cgroup configuration: unable to start unit "docker-dc614b3738760286e167a6fb3e4cd167189c59613e59e87458f7ec343b41c9f3.scope" (properties [{Name:Description Value:"libcontainer container dc614b3738760286e167a6fb3e4cd167189c59613e59e87458f7ec343b41c9f3"} {Name:Slice Value:"user.slice"} {Name:Delegate Value:true} {Name:PIDs Value:@au [2618]} {Name:MemoryAccounting Value:true} {Name:CPUAccounting Value:true} {Name:IOAccounting Value:true} {Name:TasksAccounting Value:true} {Name:DefaultDependencies Value:false}]): Interactive authentication required.: unknown

**Comment by dev6699** [comment]
Running into this issue on Ubuntu 22.04, Docker version: 23.0.1 rootless mode

Got this message `Could not determine if daemon is containerized: open /proc/1/cgroup: no such file or directory` when running `dockerd-rootless.sh` script.

Found the cause due to some hardening script(Ansible) applied on the system:
```bash
- name: Set option hidepid for proc filesystem
  ansible.posix.mount:
    path: /proc
    src: proc
    fstype: proc
    opts: rw,nosuid,nodev,noexec,relatime,hidepid=2
    state: mounted
```
Set `hidepid` back to default 0 solve the issue 
```bash
- name: Set option hidepid for proc filesystem
  ansible.posix.mount:
    path: /proc
    src: proc
    fstype: proc
    opts: rw,nosuid,nodev,noexec,relatime,hidepid=0
    state: mounted
```
Hope this help.

**Comment by ww7** [comment]
Works on DO droplet with Ubuntu 22 after:
```
echo "GRUB_CMDLINE_LINUX=systemd.unified_cgroup_hierarchy=false" | sudo tee /etc/default/grub.d/cgroup.cfg
sudo update-grub
reboot
```

**Comment by thaJeztah** [comment]
@AkihiroSuda anything actionable on this ticket, or should this one be closed?

**Comment by sam-thibault** [comment]
I will close this as stale.

**Comment by willnode** [comment]
This appears on my case because I use `hidepid` and I wouldn't want to disable it.

So I set `~/.config/docker/daemon.json` to

```json
{"exec-opts": ["native.cgroupdriver=cgroupfs"]}
```

Works like a charm! no system change needed.

**Comment by tataouinea** [comment]
For anyone encountering this `Interactive authentication required.: unknown` error while running rootless Docker on WSL, try this workaround:

```bash
sudo systemctl restart user@1000.service
```

This appears to be related to [microsoft/WSL#13143](https://github.com/microsoft/WSL/issues/13143). [Reportedly](https://github.com/microsoft/WSL/issues/13143#issuecomment-2993221016), the issue was resolved in WSL version 2.6.0, though I haven't personally tested this version yet.
